Neos.Fusion/Classes/Core/FusionSourceCodeIsInvalid.php

```diff
@@ -0,0 +1,31 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Fusion\Core;
+
+/*
+ * This file is part of the Neos.Fusion package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Fusion\Exception;
+use Neos\Flow\Annotations as Flow;
+
+#[Flow\Proxy(false)]
+class FusionSourceCodeIsInvalid extends Exception
+{
+    public static function becauseTheSourceCodeIsEmpty(?string $contextPathAndFileName): self
+    {
+        return new self("The sourcecode of the FusionSourceCode must not be empty. $contextPathAndFileName", 1657963664);
+    }
+
+    public static function becauseTheFileNameIsNotReadable(string $attemptedFileName): self
+    {
+        return new self("Trying to read Fusion source code from file, but '$attemptedFileName' is not readable.", 1657963790);
+    }
```

Comment:
I **hate** that whenever Doctrine throws an exception! Also, it makes the exception code no longer point to *the one place* the exception really comes from. Please, let us not introduce this "pattern".

Reply:
fine with me ;)

Reply:
@nezaniel just said that the new escr may or may not look entirely like this ^^

Reply:
For me I have rarely seen the need for this. Most exceptional things happen in one place. Elsewhere something else goes wrong, so why would I add a static method for that?

I'd give in to the team only if this is introduced with the code being a method argument - because that unique ID has helped tremendously over the years.

Reply:
yes in this case i did it only for fun - and because it looks cool 😎 - and i like this pattern.

i think this could easily grow into a big discussion and i agree on both sides (i have talked with some people already about this)

for this particular change, im fine with reverting.

Reply:
I'd welcome that. We should discuss this separately.

Reply:
okay lets move the discussion to https://discuss.neos.io/t/rfc-updated-default-code-style/5836?u=marc